utils/logger.go

```diff
@@ -17,12 +20,36 @@ const (
 
 // NewLogger creates and returns a new instance of the Logger type. This
 // initializes the logger with a tagged logrus Entry initialized with the
-// version string and hostname.
-func NewLogger(version string) Logger {
+// version string, hostname and two logging options (syslog & logpath)
+// If syslog is true the logger will use local syslog method
+// If syslog is false logpath will be evaluated as log file with rotation
+// STDOUT is the logging fallback method
+func NewLogger(version string, syslogEnabled bool, logpath string) Logger {
 	log := logrus.New()
+	if syslogEnabled {
+		hook, err := logrus_syslog.NewSyslogHook("", "", syslog.LOG_INFO, "device-hub")
+		if err == nil {
+			log.Formatter = &logrus.TextFormatter{
+				DisableColors: true,
+			}
+			log.Hooks.Add(hook)
+			logger := log.WithFields(defaultFields(version))
+			return &l{entry: logger}
+		}
+	}
 	log.Formatter = new(logrus.JSONFormatter)
 	log.Level = logrus.InfoLevel
-	log.Out = os.Stdout
+	// TODO Parameterize Maximums
+	if len(logpath) != 0 {
+		log.Out = &lumberjack.Logger{
+			Filename:   logpath,
+			MaxSize:    3, // Mb
+			MaxBackups: 3,
+			MaxAge:     28, // days
+		}
+	} else {
+		log.Out = os.Stdout
+	}
```

Comment:
I'm not 100% sure that this should be an "OR" choice e.g. why not write to log file and stdout?

io.MultiWriter could be helpful here